src/qmxgraph/page/utils.js

```diff
@@ -66,16 +66,58 @@ graphs.utils.setStyleKey = function setStyleKey (style, key, value) {
         }
         else
         {
-            var cont = style.indexOf(';', index);
+            var parts = style.split(';')
+            for (var i = 0; i < parts.length; ++i)
+            {
+                var current = parts[i];
+                var current_key = current.split('=')[0];
+                if (current_key == key)
+                {
+                    break; // Just to find the key index
+                }
+            }
```

Comment:
We should error out if we don't find the key in the style string.

Reply:
Or change the method name to some thing diferent... like `makeSureStyleKeyIsNoSet`...

```
>> var e = document.createElement('div')
undefined
>> e.hasAttribute('foo')
false
>> e.removeAttribute('foo')
undefined
>> null.foo
Uncaught TypeError: null has no properties
    <anonymous> debugger eval code:1
debugger eval code:1:1
    <anonymous> debugger eval code:1
```

Silently ignoring missing is not totally alien.
Maybe add an `ignore_missing` parameter, raise when false, do nothing when true.

Reply:
This is on the `setStyleKey`, which is expected to add the new key or update the existing one, so this is correct IMO.

One point is that this `else` block corresponds to updating an existing key, but because of the initial `indexOf(key+'=')`, this can be tricked by the substring issue that originated this change. But anyway this loop will end up iterating on all parts and adding the key to the end, appending a new key and resulting in the expected output.

```
>>> graphs.utils.setStyleKey('foobar=3;fizzbar=7', 'bar', 5)
'foobar=3;fizzbar=7;bar=5'
```

Reply:
My main concern is that it seemed `current_key` would point to an incorrect key if we didn't reached the `break` statement. 

The new code uses `splice` with `i`, I assume this will append the key to the end if `i == parts.length`?

Reply:
Yes, it does.